File: Sorting Algorithms/Gabri/Decrease&Conquer/InsertionSort/InsertionSort.py
def insertion_sort(arr):
    for i in range(len(arr)):
        el = arr[i]

        j = i - 1
        while (j >= 0 and arr[j] > el):
            arr[j + 1] = arr[j]
            j = j-1

        arr[j+1] = el


def main():
    numbers = [89, 45, 68, 90, 29, 34, 17]
    print(f'Unsorted: {numbers}')

    insertion_sort(numbers)
    print(f'Sorted: {numbers}')

main()

File: Sorting Algorithms/Gabri/Decrease&Conquer/InsertionSort/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {
    private static void insertionSort(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            int el = arr[i];

            int j = i - 1;
            while (j >= 0 && arr[j] > el) arr[j + 1] = arr[j--];

            arr[j+1] = el;
        }
    }

    public static void main(String[] args) {
        int[] numbers = { 89, 45, 68, 90, 29, 34, 17 };
        System.out.println("Unsorted: " + Arrays.toString(numbers));

        insertionSort(numbers);
        System.out.println("Sorted: " + Arrays.toString(numbers));
    }
}
